store/organizations.js

```diff
@@ -23,6 +28,29 @@ const $mutations = {
 
   [UPDATE_CACHE](state, organizations = []) {
     state.organizationCache = _.unionBy(organizations, state.organizationCache, (o) => o.organizationId);
+    _.forEach(organizations, (organization) => {
+      if (organization && organization.organizationId) {
+        const index = state.selectedOrganizations.findIndex((o) => o.organizationId === organization.organizationId);
+        if (index >= 0) state.selectedOrganizations.splice(index, 1, organization);
+      }
```

Comment:
remove this if condition there. you cannot pass and array without organization inside. this check prevent good code elsewhere

add separate validation check at line 30 that test that the passed array !=null * evry value are valid. if not throw error.

eg: line 30
```javascript
// 1st line of the Methods (line 30)
const isValid = organizations && organizations.every((o)=> o && o.orgnizationId);
if(!isValid) throw Error("organizations list is invalid");
```

then do operations

 ```suggestion
      const index = state.selectedOrganizations.findIndex((o) => o.organizationId === organization.organizationId);
      if (index >= 0) state.selectedOrganizations.splice(index, 1, organization);
```

Reply:
Hi @sbilodeau, Added separate validation and removed loop. 

Had done this because we don't require to loop through organizations array twice (One for validate, one for actual operation).